src/FS.Embedded/Manifest/EmbeddedFilesManifest.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    internal class EmbeddedFilesManifest
+    {
+        private static readonly char[] _invalidFileNameChars = Path.GetInvalidFileNameChars()
+            .Where(c => c != Path.DirectorySeparatorChar && c != Path.AltDirectorySeparatorChar).ToArray();
+
+        private static readonly char[] Separators = new char[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar };
+        private readonly ManifestDirectory _rootDirectory;
+
+        internal EmbeddedFilesManifest(ManifestDirectory rootDirectory)
+        {
+            if (rootDirectory == null)
+            {
+                throw new ArgumentNullException(nameof(rootDirectory));
+            }
+
+            _rootDirectory = rootDirectory;
+        }
+
+        internal ManifestEntry ResolveEntry(string path)
+        {
+            if (string.IsNullOrEmpty(path) || HasInvalidPathChars(path))
+            {
+                return null;
+            }
+
+            // trimmed is a string without leading nor trailing path separators
+            // so if we find an empty string while iterating over the segments
+            // we know for sure the path is invalid and we treat it as the above
+            // case by returning null.
+            // Examples of invalid paths are: //wwwroot /\wwwroot //wwwroot//jquery.js
+            var trimmed = RemoveLeadingAndTrailingDirectorySeparators(path);
+            // Paths consisting only of a single path separator like / or \ are ok.
+            if (trimmed.Length == 0)
+            {
+                return _rootDirectory;
+            }
+
+            var tokenizer = new StringTokenizer(trimmed, Separators);
+            ManifestEntry currentEntry = _rootDirectory;
+            foreach (var segment in tokenizer)
+            {
+                if (segment.Equals(""))
+                {
+                    return null;
```

Comment:
Does this mean that the path was an empty string or something like "./"?

Reply:
If so, shouldn't this return the _rootDirectory?

Reply:
This means that there were two separators together on the path, like "//" or like "wwwroot//site.css"
See:
https://github.com/aspnet/FileSystem/pull/310/files#diff-841c910febd853df519b2ff09175ed0cR35
https://github.com/aspnet/FileSystem/pull/310/files#diff-841c910febd853df519b2ff09175ed0cR13

Reply:
./ will return _rootDirectory